### DISCUSSION TITLE
How do the comments in the database table correspond to the fields?

### DISCUSSION THREAD

**Discussion by darvsum** [vanna-ai/vanna#206] [Q&A]
How do the comments in the database table correspond to the fields?
Since the meaning of comments in the database table is usually used when describing the problem, the generated SQL cannot find the field.

**Comment by zainhoda** [comment]
Could you provide an example of how you've trained it?

**Reply by darvsum** [reply]
vn.train(ddl="""
   CREATE TABLE `test`.`t_yangxin`  (
  `region_code` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NULL DEFAULT NULL COMMENT 'region code',
  `region_name` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NULL DEFAULT NULL COMMENT 'region name',
  `dept_name` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NULL DEFAULT NULL COMMENT 'department name',
  `transact_name` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NULL DEFAULT NULL COMMENT 'matter name',
  `task_name` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NULL DEFAULT NULL COMMENT 'task name',
  `transact_code` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NOT NULL COMMENT 'matter code',
  `transact_type` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NULL DEFAULT NULL COMMENT 'matter type',
  `transact_status` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NULL DEFAULT NULL COMMENT 'matter status',
  PRIMARY KEY (`transact_code`) USING BTREE
) ENGINE = InnoDB CHARACTER SET = utf8mb4 COLLATE = utf8mb4_general_ci COMMENT = 'Yangxin handling matter' ROW_FORMAT = Dynamic;
""")

vn.ask("How many pieces of data are there in t_yangxin whose matter type is public service?")

getting the wrong answer：
df = pd.read_sql_query(sql, conn)
Couldn't run sql:  Execution failed on sql 'SELECT COUNT(*)
FROM t_yangxin
WHERE matter_type = 'Public Service';': (1054, "Unknown column 'matter_type' in 'where clause'")
